prov/efa/src/rdm/rxr_ep.c

```diff
@@ -843,12 +800,13 @@ void rxr_ep_set_use_shm_for_tx(struct rxr_ep *ep)
 
 	/* App provided hints supercede environmental variables.
 	 *
-	 * Using the shm provider comes with some overheads, particularly in the
-	 * progress engine when polling an empty completion queue, so avoid
+	 * Using the shm provider comes with some overheads, so avoid
 	 * initializing the provider if the app provides a hint that it does not
 	 * require node-local communication. We can still loopback over the EFA
 	 * device in cases where the app violates the hint and continues
 	 * communicating with node-local peers.
+	 *
+	 * aws-ofi-nccl relies on this feature.
```

Comment:
Is this comment from another commit?

Reply:
I could happen to add this comment when I rebase. Let me look at it.

Reply:
Yes, I added this line. The comment is about why we don't enable SHM by default. The reason provided is that the additional CQ polling adds additional overhead when SHM is enabled. With peer API, that's no longer a valid reason. However, we still cannot enable SHM by default because OFI NCCL plugin does not want SHM.